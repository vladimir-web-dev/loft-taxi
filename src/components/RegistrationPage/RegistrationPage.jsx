import React from 'react';
import RegistrationForm from '../RegistrationForm';


import logo from '../../images/logo.png';
import './RegistrationPage.css'

export const RegistrationPage = () => {
    return (
        <section className='section registration'>
            <div className='registration__content'>
                    <div className='registration__left'> 
                        <img className='logo' src={logo} alt='Logo'></img>
                    </div>
                    <div className='registration__right'>
                        <RegistrationForm />
                    </div>        
                </div>                       
        </section>
    );   
};
