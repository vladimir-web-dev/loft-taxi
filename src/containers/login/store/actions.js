import { createAction } from 'redux-actions';

export const authRequest = createAction('taxi/auth/AUTH_REQUEST');
export const authSuccess = createAction('taxi/auth/AUTH_SUCCESS');
export const authFailure = createAction('taxi/auth/AUTH_FAILURE');

export const registrationRequest = createAction('taxi/auth/REGISTRATION_REQUEST');
export const registrationSuccess = createAction('taxi/auth/REGISTRATION_SUCCESS');
export const registrationFailure = createAction('taxi/auth/REGISTRATION_FAILURE');

export const logout = createAction('taxi/auth/LOGOUT');

