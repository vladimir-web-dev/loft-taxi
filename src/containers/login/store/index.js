export { default } from './reducer';
export * from './actions';
export * from './selectors';
export * from './middlewares';
export * from './sagas';