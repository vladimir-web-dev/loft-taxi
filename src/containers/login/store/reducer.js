import {
    authRequest,
    authSuccess,
    authFailure,
    registrationRequest,
    registrationSuccess,
    registrationFailure,
    logout
} from './actions';
import { handleActions } from 'redux-actions';
import { combineReducers } from 'redux';

const token = handleActions({
    [authRequest]: () => "",
    [authSuccess]: (_state, action) => action.payload,
    [registrationRequest]: () => "",
    [registrationSuccess]: (_state, action) => action.payload,
    [logout]: () => ""
}, "");

const isAuthenticating = handleActions({
    [authRequest]: () => true,
    [authSuccess]: () => false,
    [authFailure]: () => false,
    [registrationRequest]: () => true,
    [registrationSuccess]: () => false,
    [registrationFailure]: () => false
}, false);

const isAuthenticated = handleActions({   
    [authSuccess]: () => true,
    [authFailure]: () => false,  
    [registrationSuccess]: () => true,
    [registrationFailure]: () => false,
    [logout]: () => false
}, false);

const authError = handleActions({   
    [authSuccess]: () => "",
    [authFailure]: (_state, action) => action.payload,  
}, "");

const registrationError = handleActions({    
    [registrationSuccess]: () => "",
    [registrationFailure]: (_state, action) => action.payload
}, "");

export default combineReducers({isAuthenticating, isAuthenticated, token, authError, registrationError});
