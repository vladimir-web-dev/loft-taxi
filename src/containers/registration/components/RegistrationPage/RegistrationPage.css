.registration {   
    background-size: cover;
    background-position: center;
    background-image: 
        linear-gradient(rgba(0,0,0, .3), rgba(0,0,0, .4)),
        url('../../../../images/background.png');
}

.registration__content {
    max-width: 900px;
    width: 900px;
}

.registration,
.registration__content {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
}

.registration__left,
.registration__rght {
    flex: 1;
}