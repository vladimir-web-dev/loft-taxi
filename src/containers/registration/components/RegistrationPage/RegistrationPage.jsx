import React from "react";
import { RegistrationForm } from "../RegistrationForm";
import { Logo } from "loft-taxi-mui-theme";
import containerStyles from  "../../../../css_modules/Container.module.css";

export const RegistrationPage = ({ history }) => {
  return (
    <section className={containerStyles.central}>
      <div className={containerStyles.content}>
        <div className={containerStyles.col}>
          <Logo />
        </div>
        <div className={containerStyles.col}>
          <RegistrationForm history={history} />
          </div>
      </div>
    </section>
    
  );
};
