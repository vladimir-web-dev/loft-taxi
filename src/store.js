import { createStore, applyMiddleware, compose } from 'redux';
import authReducer, { mwAuthentication, mwRegistration } from './modules/auth';
import cardReducer, { mwCard } from './modules/card';
import { combineReducers } from 'redux';


export const initStore = () => { 
    return createStore(
        combineReducers({
            authReducer, 
            cardReducer
        }),
        compose(
            applyMiddleware(mwAuthentication),
            applyMiddleware(mwRegistration),
            applyMiddleware(mwCard)
        )
    );
};
    