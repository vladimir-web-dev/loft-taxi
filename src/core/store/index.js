import { createStore, applyMiddleware, compose } from "redux";
import { rootReducer } from "../reducer";
import { rootSaga } from "../saga";
import createSagaMiddleware from "redux-saga";

export function initStore() {
  const sagaMiddleware = createSagaMiddleware();
  const store = createStore(rootReducer, compose(
    applyMiddleware(sagaMiddleware)
    ));

  sagaMiddleware.run(rootSaga);

  return store;
}
