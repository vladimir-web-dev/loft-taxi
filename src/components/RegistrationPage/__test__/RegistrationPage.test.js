import React from "react";
import ReactDOM from "react-dom";
import { RegistrationPage } from "../RegistrationPage";
import { render } from "@testing-library/react";
import { initStore } from "../../../store";
import { Provider } from "react-redux";
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";
import "mutationobserver-shim";
global.MutationObserver = window.MutationObserver;


it("renders without crashing", () => {
  const store = initStore();
  const div = document.createElement("div");
  const history = createMemoryHistory();

  ReactDOM.render(
    <Router history={history}>
      <Provider store={store}>
        <RegistrationPage />
      </Provider>
    </Router>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it("matches snapshot", () => {
  const store = initStore();
  const div = document.createElement("div");
  const history = createMemoryHistory();

  const { container } = render(
    <Router history={history}>
      <Provider store={store}>
        <RegistrationPage />
      </Provider>
    </Router>,
    div
  );
  expect(container).toMatchSnapshot();
});
