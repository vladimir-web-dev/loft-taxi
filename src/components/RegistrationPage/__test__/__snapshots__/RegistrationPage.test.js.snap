// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <section
    class="section registration"
  >
    <div
      class="registration__content"
    >
      <div
        class="registration__left"
      >
        <img
          alt="Logo"
          class="logo"
          src="logo.png"
        />
      </div>
      <div
        class="registration__right"
      >
        <form
          class="formTag"
          data-testid="registration-form"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
            elevatio="1"
            style="padding: 48px 48px;"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              >
                <h1
                  class="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom"
                >
                  Регистрация
                </h1>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              >
                Уже зарегистрированы?
                <a
                  data-testid="link-login"
                  href="/login"
                >
                   Войти
                </a>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
                  data-testid="input-email"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense Mui-required Mui-required"
                    data-shrink="false"
                  >
                    Адрес электронной почты
                    <span
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense MuiInput-marginDense"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
                      placeholder="Адрес электронной почты"
                      required=""
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-grid-xs-12"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
                >
                  <div
                    class="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
                    data-testid="input-fname"
                  >
                    <label
                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense Mui-required Mui-required"
                      data-shrink="false"
                    >
                      Имя
                      <span
                        class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                      >
                         
                        *
                      </span>
                    </label>
                    <div
                      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense MuiInput-marginDense"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
                        name="name"
                        placeholder="Имя"
                        required=""
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
                >
                  <div
                    class="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
                    data-testid="input-lname"
                  >
                    <label
                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense Mui-required Mui-required"
                      data-shrink="false"
                    >
                      Фамилия
                      <span
                        class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                      >
                         
                        *
                      </span>
                    </label>
                    <div
                      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense MuiInput-marginDense"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
                        name="surname"
                        placeholder="Фамилия"
                        required=""
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
                  data-testid="input-pass"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense Mui-required Mui-required"
                    data-shrink="false"
                  >
                    Пароль
                    <span
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense MuiInput-marginDense"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
                      name="password"
                      placeholder="Пароль"
                      required=""
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-xs-flex-end MuiGrid-grid-xs-12"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
                  data-testid="button-login"
                  tabindex="0"
                  type="submit"
                >
                  <span
                    class="MuiButton-label"
                  >
                    Войти
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </section>
</div>
`;
